spotify-bin

```diff
@@ -43,11 +44,18 @@ if [ $URI_HANDLED -eq 0 ]; then
 fi
 
 if [ -f "${XDG_CONFIG_HOME}/spotify-flags.conf" ]; then
-    mapfile -t SPOTIFY_USER_FLAGS <<< "$(grep -v '^#' "${XDG_CONFIG_HOME}/spotify-flags.conf")"
+    mapfile -t EXTRA_FLAGS <<< "$(grep -v '^#' "${XDG_CONFIG_HOME}/spotify-flags.conf")"
 fi
 
-env PULSE_PROP_application.icon_name="com.spotify.Client" LD_PRELOAD=/app/lib/spotifywm.so:/app/lib/spotify-preload.so${LD_PRELOAD:+:$LD_PRELOAD} /app/extra/bin/spotify --force-device-scale-factor=$SCALE_FACTOR "${SPOTIFY_USER_FLAGS[@]}" "$@" &
-`set-dark-theme-variant.py` &
+if [ "${SCALE_FACTOR}" != "1.0" ]; then
+    EXTRA_FLAGS+=("--force-device-scale-factor=${SCALE_FACTOR}")
+fi
+
+if [[ -e "${XDG_RUNTIME_DIR}/${WAYLAND_SOCKET}" || -e "${WAYLAND_DISPLAY}" ]]; then
+    EXTRA_FLAGS+=("--enable-wayland-ime" "--ozone-platform-hint=auto")
```

Comment:
iirc it's not default because there are still some issues. I'm not sure if forcing this is right. Maybe it should be left for users to decide.

Reply:
This isn't the default, by default the Wayland socket is not available to Spotify, the user has to go out of their way to enable it.

Reply:
maybe the socket should be enabled by default but the flags will be left for users to choose? That would match non-flatpak behavior which more users may be familiar with.

Reply:
The flags file is a custom flatpak change anyways, so they'd need to have read this repo's README anyways to know about it.

Also this way if any additional flags are needed for Wayland in the future, we can add them here instead of every user needing to modify their flags file.

Though, enabling the socket by default may be nice as it'll auto switch when they turn on support upstream.

Reply:
I like how it was done here. Other packages do the same.